monitoring/port.go

```diff
@@ -104,14 +107,20 @@ func (c *portChecker) checkProcess(proc process, reporter health.Reporter) bool
 		if r.Protocol != proc.socket.proto() {
 			continue
 		}
-		// ignore sockets in time-wait and closed states since they're going
-		// away soon
 		switch proc.socket.state() {
 		case TimeWait:
+			// ignore sockets in time-wait and closed states since they're going
+			// away soon
 			log.Debugf("Ignoring %v for program %q(pid=%v).", formatSocket(proc.socket), proc.name, proc.pid)
 			continue
 		}
-		if uint64(proc.localAddr().port) >= r.From && uint64(proc.localAddr().port) <= r.To {
+		localAddr := proc.localAddr().ip.String()
+		addr := r.ListenAddr
+		if addr == "" {
+			addr = localAddr
+		}
+		if uint64(proc.localAddr().port) >= r.From && uint64(proc.localAddr().port) <= r.To &&
+			localAddr == addr {
```

Comment:
I think this will fail to detect services bound to any address, ie 0.0.0.0, which I believe will conflict with a bind to a specific address.

Reply:
Good catch, Kevin